src/export/dk.ml

```diff
@@ -0,0 +1,183 @@
+let current_path = Stdlib.ref []
+
+let path : Path.t pp = fun ppf p ->
+  if p <> Stdlib.(!current_path) then
+  match p with
+  | [] -> ()
+  | [_root_path; m] -> out ppf "%a." ident m
+  | _ -> assert false
+
+(** Type of Dedukti declarations. *)
+type decl =
+  | Sym of sym
+  | Rule of (Path.t * string * rule)
+
+(** Declarations are ordered wrt their positions in the source. *)
+
+let pos_of_decl : decl -> Pos.popt = fun i ->
+  match i with
+  | Sym s -> s.sym_pos
+  | Rule (_,_,r) -> r.rule_pos
+
+let cmp : decl cmp = cmp_map (Lplib.Option.cmp Pos.cmp) pos_of_decl
+
+(** Translation of terms. *)
+
+let tvar : tvar pp = fun ppf v -> ident ppf (Bindlib.name_of v)
+let tevar : tevar pp = fun ppf v -> ident ppf (Bindlib.name_of v)
+
+let rec term : term pp = fun ppf t ->
+  match unfold t with
+  | Vari v -> tvar ppf v
+  | Type -> out ppf "Type"
+  | Kind -> assert false
+  | Symb s -> out ppf "%a%a" path s.sym_path ident s.sym_name
+  | Prod(a,b) ->
+    let x,b' = Bindlib.unbind b in
+    if Bindlib.binder_constant b then out ppf "(%a -> %a)" term a term b'
+    else out ppf "(%a : %a -> %a)" tvar x term a term b'
+  | Abst(a,b) ->
+    let x,b = Bindlib.unbind b in
+    out ppf "(%a : %a => %a)" tvar x term a term b
+  | Appl _ ->
+    let h, ts = get_args t in
+    out ppf "(%a%a)" term h (List.pp (prefix " " term) "") ts
+  | LLet(a,t,u) ->
+    let x,u = Bindlib.unbind u in
+    out ppf "((%a : %a := %a) => %a)" tvar x term a term t term u
+  | Patt(_,s,ts) -> out ppf "%a%a" ident s (Array.pp (prefix " " term) "") ts
+  | TEnv(te, ts) -> out ppf "%a%a" tenv te (Array.pp (prefix " " term) "") ts
+  | TRef _ -> assert false
+  | Wild -> assert false
+  | Meta _ -> assert false
+
+and tenv : term_env pp = fun ppf te ->
+  match te with
+  | TE_Vari v -> tevar ppf v
+  | TE_Some _ -> assert false
+  | TE_None -> assert false
+
+(** Translation of declarations. *)
+
+let modifiers : sym -> string list = fun s ->
+  let open Stdlib in
+  let r = ref [] in
+  let add m = r := m::!r in
+  if s.sym_opaq then add "thm"
+  else begin
+    if s.sym_expo = Protec then add "private";
+    match s.sym_prop with
+    | Const -> ()
+    | Injec -> add "injective"
+    | AC _ -> add "defac"
+    | Defin -> add "def"
+    | Assoc _ -> assert false
+    | Commu -> assert false
+  end;
+  !r
+
+let sym_decl : sym pp = fun ppf s ->
+  out ppf "%a%a : %a%a.@."
+    (List.pp (suffix string " ") "") (modifiers s)
+    ident s.sym_name term !(s.sym_type)
+    (Option.pp (prefix " :=" term)) !(s.sym_def)
+
+let rule_decl : (Path.t * string * rule) pp = fun ppf (p, n, r) ->
+  let xs, rhs = Bindlib.unmbind r.rhs in
+  out ppf "[%a] %a%a%a --> %a.@."
+    (Array.pp tevar ", ") xs
+    path p ident n (List.pp (prefix " " term) "") r.lhs term rhs
+
+let decl : decl pp = fun ppf decl ->
+  match decl with
+  | Sym s -> sym_decl ppf s
+  | Rule r -> rule_decl ppf r
+
+(** [decls_of_sign sign] computes an ordered list of declarations from the
+   signature [sign]. *)
```

Comment:
I think it would be wise to state in the comments that the list is ordered by the position of the declaration in the source file.